### PR TITLE
Add missing configuration for DEFAULT_ACCEPTMBOX_SIZE (fix #12509) (IDFGH-11364)

### LINKED ISSUE #12509: LWIP's DEFAULT_ACCEPTMBOX_SIZE is not configurable (IDFGH-11363)
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.2-dev-3775-gb4268c874a

### Espressif SoC revision.

n/a

### Operating System used.

macOS

### How did you build your project?

Command line with idf.py

### If you are using Windows, please specify command line type.

None

### Development Kit.

n/a

### Power Supply used.

USB

### What is the expected behavior?

It is expected that, provided proper configuration, the backlog of listening TCP/IP sockets is honored.

For example, when listening on a TCP/IP socket with a backlog of 10, we do expect 10 clients to be able to connect simultaneously.

### What is the actual behavior?

The actual processing of the backlog is limited to DEFAULT_ACCEPTMBOX_SIZE which, unlike DEFAULT_TCP_RECVMBOX_SIZE, is hard coded to 6 and not configurable in `lwipopts.h`

As a result, 4 clients receive RST packets.

### Steps to reproduce.

Create a TCP/IP socket, bind it, listen with a backlog of 10.
Start connecting with ten clients at once from another host.
Notice the 4 RST packets.

### Debug Logs.

_No response_

### More Information.

_No response_

### PR DESCRIPTION
**Pull request by pguyot**

### CODE REVIEW

File: components/lwip/Kconfig [hunk 1]
```diff
@@ -657,6 +657,20 @@ menu "LWIP"
                 mail box is full, the LWIP drops the packets. So generally we need to make sure the TCP
                 receive mail box is big enough to avoid packet drop between LWIP core and application.
 
+        config LWIP_ACCEPTMBOX_SIZE
```
Comment (espressif-abhikroy) [inline components/lwip/Kconfig:660]:
```suggestion
        config LWIP_TCP_ACCEPTMBOX_SIZE
```

File: components/lwip/port/include/lwipopts.h [hunk 2]
```diff
@@ -834,7 +834,7 @@ static inline uint32_t timeout_from_offered(uint32_t lease, uint32_t min)
  * The queue size value itself is platform-dependent, but is passed to
  * sys_mbox_new() when the acceptmbox is created.
  */
-#define DEFAULT_ACCEPTMBOX_SIZE         6
+#define DEFAULT_ACCEPTMBOX_SIZE         CONFIG_LWIP_ACCEPTMBOX_SIZE
```
Comment (espressif-abhikroy) [inline components/lwip/port/include/lwipopts.h:837]:
```suggestion
#define DEFAULT_ACCEPTMBOX_SIZE         CONFIG_LWIP_TCP_ACCEPTMBOX_SIZE
```

File: tools/ldgen/samples/sdkconfig [hunk 3]
```diff
@@ -362,6 +362,7 @@ CONFIG_LWIP_TCP_MSL=60000
 CONFIG_LWIP_TCP_SND_BUF_DEFAULT=5744
 CONFIG_LWIP_TCP_WND_DEFAULT=5744
 CONFIG_LWIP_TCP_RECVMBOX_SIZE=6
+CONFIG_LWIP_ACCEPTMBOX_SIZE=6
```
Comment (espressif-abhikroy) [inline tools/ldgen/samples/sdkconfig:365]:
```suggestion
CONFIG_LWIP_TCP_ACCEPTMBOX_SIZE=6
```

File: components/esp_wifi/src/wifi_init.c [hunk 4]
```diff
@@ -184,6 +184,7 @@ static void esp_wifi_config_info(void)
 #endif
 
 #ifdef CONFIG_ESP_NETIF_TCPIP_LWIP
+    ESP_LOGI(TAG, "accept mbox: %d", CONFIG_LWIP_ACCEPTMBOX_SIZE);
```
Comment (espressif-abhikroy) [inline components/esp_wifi/src/wifi_init.c:187]:
```suggestion
    ESP_LOGI(TAG, "accept mbox: %d", CONFIG_LWIP_TCP_ACCEPTMBOX_SIZE);
```

**Comment by Alvin1Zhang** [comment]
Thanks for contribution again, changes have been merged with https://github.com/espressif/esp-idf/commit/336c0946fa3c7fdbbaeecfea4056eb57c0917c2c.

**Pull request closed without merging**